nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/util/SecureNiFiConfigUtil.java

```diff
@@ -0,0 +1,191 @@
+        }
+    }
+
+    /**
+     * Attempts to add some reasonable guesses at desired SAN values that can be added to the generated
+     * certificate.
+     * @param nifiProperties The nifi.properties
+     * @return A Pair with IP SANs on the left and DNS SANs on the right
+     */
+    private static Pair<String[], String[]> getSubjectAlternativeNames(Properties nifiProperties) {
+        Set<String> ipSubjectAlternativeNames = new HashSet<>();
+        Set<String> dnsSubjectAlternativeNames = new HashSet<>();
+
+        try {
+            ipSubjectAlternativeNames.add(InetAddress.getLocalHost().getHostAddress());
+            dnsSubjectAlternativeNames.add(InetAddress.getLocalHost().getHostName());
+        } catch (UnknownHostException e) {
+            // This was just a courtesy, so we'll skip adding it
+        }
+        addSubjectAlternativeName(nifiProperties, NiFiProperties.REMOTE_INPUT_HOST, ipSubjectAlternativeNames, dnsSubjectAlternativeNames);
+        addSubjectAlternativeName(nifiProperties, NiFiProperties.WEB_HTTPS_HOST, ipSubjectAlternativeNames, dnsSubjectAlternativeNames);
+        addSubjectAlternativeName(nifiProperties, NiFiProperties.WEB_PROXY_HOST, ipSubjectAlternativeNames, dnsSubjectAlternativeNames);
+        addSubjectAlternativeName(nifiProperties, NiFiProperties.LOAD_BALANCE_ADDRESS, ipSubjectAlternativeNames, dnsSubjectAlternativeNames);
+
+        // Not necessary to add as a SAN
+        ipSubjectAlternativeNames.remove(LOCALHOST_IP);
+        dnsSubjectAlternativeNames.remove(LOCALHOST_NAME);
+
+        return new ImmutablePair<>(ipSubjectAlternativeNames.toArray(new String[ipSubjectAlternativeNames.size()]),
+                dnsSubjectAlternativeNames.toArray(new String[dnsSubjectAlternativeNames.size()]));
+    }
+
+    private static void addSubjectAlternativeName(Properties nifiProperties, String propertyName, Set<String> ipSubjectAlternativeNames,
+                                                  Set<String> dnsSubjectAlternativeNames) {
+        String hostValue = nifiProperties.getProperty(propertyName, "");
+        if (!hostValue.isEmpty()) {
+            if (IPAddress.isValid(hostValue)) {
+                ipSubjectAlternativeNames.add(hostValue);
+            } else {
+                dnsSubjectAlternativeNames.add(hostValue);
+            }
+        }
+    }
+
+    private static void updateProperties(final File propertiesFile, final TlsConfiguration tlsConfiguration) throws IOException {
+        final Path propertiesFilePath = propertiesFile.toPath();
+        final List<String> lines = Files.readAllLines(propertiesFilePath);
+        final List<String> updatedLines = lines.stream().map(line -> {
+            if (line.startsWith(NiFiProperties.SECURITY_KEYSTORE_PASSWD)) {
+                return String.format("%s=%s", NiFiProperties.SECURITY_KEYSTORE_PASSWD, tlsConfiguration.getKeystorePassword());
```

Comment:
Recommend declaring `%s=%s` as a private static variable and reusing it across the lines, or adding a simple `getPropertyLine()` method to wrap the String formatting.